### ISSUE TITLE
Add @CartesianEnumSource

### ISSUE THREAD

**Issue by scordio** [junit-pioneer/junit-pioneer#379]
I am currently looking for a way to provide Enum values as input of a cartesian product test.

I understand it can be achieved with a [static factory method](https://junit-pioneer.org/docs/cartesian-product/#writing-a-static-factory-method-for-the-parameters), but having something like a `@CartesianEnumSource` which mimics the JUnit `@EnumSource` including `names` and `mode` parameters would be really nice.

If this sounds reasonable, I am happy to give it a shot (probably taking inspiration from #321).

----

Bonus topic: it would be also nice to have a `@CartesianArgumentSource` which would allow plugging in a custom `ArgumentProvider` implementation. Maybe this deserves a separate issue, but I'm not sure yet about its feasibility.

**Comment by Michael1993** [comment]
How `CartesianProductTest` interacts with arguments sources [will change soon](https://github.com/junit-pioneer/junit-pioneer/pull/362). Once that is done, I will update the documentation to reflect these changes - i.e.: How you can write your own `ArgumentsProvider` for `CartesianProductTest` using API provided by Pioneer (it's almost like with JUnit just slightly different interfaces). This should also make implementing `@CartesianEnumSource` fairly trivial.

The issue as it is, is kind of blocked by the coming change, unfortunately. If you don't need this feature urgently, this will probably be implemented in about a month or so. Does that work?

**Comment by scordio** [comment]
Thanks for the quick feedback, good to know that something in this direction was already on the table 🙂

I would definitely wait for the change, but please do let me know if there is anything that can be picked up to speed it up.

**Comment by Michael1993** [comment]
Feel free to [review the PR](https://github.com/junit-pioneer/junit-pioneer/pull/362) implementing the change. All feedback is appreciated.

**Comment by scordio** [comment]
@Michael1993 now that #362 is merged, I would be happy to put something together to support `@CartesianEnumSource`, just let me know if I can go ahead with it.

**Comment by Michael1993** [comment]
Sure, feel free to put something together!
